routing/routing_session.cpp

```diff
@@ -577,6 +580,42 @@ shared_ptr<Route> const RoutingSession::GetRoute() const
   return m_route;
 }
 
+void RoutingSession::EnableTraffic(bool enable)
+{
+  threads::MutexGuard guard(m_routeSessionMutex);
```

Comment:
Strongly suggest you (not in this PR) to get rid of this old fashioned MutexGuards and Mutexes and switch to new and fancy lock_guards and mutexes.